userapi/src/main/java/fk/prof/userapi/UserapiConfigManager.java

```diff
@@ -61,73 +43,54 @@ public static void setDefaultSystemProperties() {
     properties.computeIfAbsent("vertx.metrics.options.enabled", k -> true);
   }
 
-  JsonObject getS3Config() {
-    JsonObject s3Config = getStorageConfig().getJsonObject(S3);
-    if(!s3Config.containsKey("list.objects.timeout.ms")) {
-      // put default value for listObject timeout
-      s3Config.put("list.objects.timeout.ms", 5000L);
-    }
-
-    // check for consistent config
-    Long requestTimeout = getUserapiHttpDeploymentConfig().getJsonObject("config").getLong("req.timeout");
-    Long ListObjectTimeout = s3Config.getLong("list.objects.timeout.ms");
-    if(requestTimeout <= ListObjectTimeout) {
-      throw new RuntimeException("request timeout must be greater than listObject timeout");
-    }
-    return s3Config;
+  Configuration.S3Config getS3Config() {
+    return getStorageConfig().s3Config;
   }
 
-
-  JsonObject getStorageThreadPoolConfig() {
-    JsonObject tpConfig = getStorageConfig().getJsonObject(THREAD_POOL);
-    checkNotEmpty(tpConfig, "thread pool");
-    return tpConfig;
+  Configuration.FixedSizeThreadPoolConfig getStorageThreadPoolConfig() {
+    return getStorageConfig().tpConfig;
   }
 
-  JsonObject getStorageConfig() {
-    JsonObject storageConfig = config.getJsonObject(STORAGE);
-    checkNotEmpty(storageConfig, "storage");
-    return storageConfig;
-  }
+  Configuration.StorageConfig getStorageConfig() {
+    Configuration.StorageConfig storageConfig = config.storageConfig;
 
-  JsonObject getBufferPoolConfig() {
-    JsonObject poolConfig = config.getJsonObject(BUFFER_POOL_OPTIONS_KEY, new JsonObject());
-    if (poolConfig.getInteger("max.total") <= 0 || poolConfig.getInteger("max.idle") < 0 || poolConfig.getInteger("buffer.size") <= 0) {
-      throw new RuntimeException("buffer pool config is not proper");
+    // check for consistent config
+    Long requestTimeout = getUserapiHttpConfig().requestTimeout;
+    Long ListObjectTimeout = storageConfig.s3Config.listObjectsTimeoutMs;
+    if(requestTimeout <= ListObjectTimeout) {
```

Comment:
Maybe have all validations in one place? => preferably constructor
It is easy to miss a validation until its too late and the service is deployed and running if validation is done in a getter (imagine a scenario in which a config value's getter is not called at bootup)